### ISSUE TITLE
Python 3 support

### ISSUE THREAD

**Issue by ShayNehmad** [guardicore/monkey#393]
**THIS ISSUE WILL TRACK THE PYTHON 3 STATUS OF THE PROJECT.**

## Expected Behavior
The Infection Monkey and the Island need to support Python 3

## Actual Behavior
Currently the project is in Python 2.

**Comment by danielguardicore** [comment]
caniusepython3 results

tldr: 2 dependencies will require work and 2 seem like they will be fine. 


project | migration status
-- | --
flask-jwt | This project is currently abandoned and will not migrate.We should use https://github.com/vimalloc/flask-jwt-extended which looks much more alive and has python 2 support as well so we can do this now
impacket | On the readme they say they support Python 3.6I'm trying to clarify it here https://github.com/SecureAuthCorp/impacket/issues/654
rdpy | In planning, but doesn't look like it's happening right now. https://github.com/citronneur/rdpy/issues/91it depends on qt4reactor which looks totally dead. https://github.com/ghtdak/qtreactor/issues/23
wmi | Readme says it supports 3.4, but the project looks kind of dead. Clarification: https://github.com/tjguk/wmi/issues/7

**Comment by danielguardicore** [comment]
#409 solved the RDPY.

**Comment by danielguardicore** [comment]
wmi works out of the box.

**Comment by ShayNehmad** [comment]
What's required to finish this feature and merge it to `develop`:

- [ ] Pass all BB tests (@VakarisZ)
- [ ] Finish all manual tests of Island and Agent features from [the test plan](https://github.com/guardicore/monkey/wiki/Feature-list-(and-testing-status)).
- - [ ] AGENT: Process list windows (@ShayNehmad)
- - [ ] AGENT: Process list linux (@ShayNehmad)
- - [ ] AGENT: Local network interface windows (@ShayNehmad)
- - [ ] AGENT: Local network interface linux (@ShayNehmad)
- - [ ] AGENT: Azure guest agent password stealing
- - [ ] AGENT: Windows service discovery (@ShayNehmad)
- - [ ] Island: AWS integration features (@ShayNehmad)
- [ ] Fix found bugs (in new issues + branches if possible)
- - [ ] Ring bug (@youknowone)
- - [ ] Ring work-around (@VakarisZ)
- - [ ] mssqlexec error handling (@VakarisZ)  
- [ ] Last CR (@ShayNehmad) 
- [ ] Update Travis CI 
- - [ ] linting 
- - [ ] pytest
- [ ] Deploy scripts (@VakarisZ)
- [ ] Merge 🍾🎉

**Comment by ShayNehmad** [comment]
AWS SSM integration works, Security Hub isn't working for me but it might be just something I'm missing. Anyway - doesn't seem critical for this version.
